delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -479,3 +478,319 @@ def handle_xml_to_dict(xml_info, element):
+                            "description": host.get('durable-id'),
+                            "storage_id": storage_id,
+                            "native_storage_host_id":
+                                host.get('serial-number'),
+                            "os_type": os_type,
+                            "status": status
+                        }
+                        host_list.append(host_dict)
+            return host_list
+        except Exception as e:
+            LOG.error("Failed to get host from msa")
+            raise e
+
+    def list_storage_host_groups(self, storage_id):
+        try:
+            host_groups_info = self.ssh_pool.do_exec('show host-groups')
+            storage_host_groups_result = {}
+            host_group_list = []
+            storage_host_grp_relation_list = []
+            host_groups = self.handle_xml_to_json(
+                host_groups_info, 'host-group')
+            host_info_list = self.handle_xml_to_json(host_groups_info, 'host')
+            if host_groups:
+                for host_group in host_groups:
+                    hosts_list = []
+                    storage_host_group_id = host_group.get('serial-number')
+                    if host_info_list:
+                        for host_info in host_info_list:
+                            host_id = host_info.get('serial-number')
+                            host_group_id = host_info.get('host-group')
+                            if host_id != 'NOHOST':
+                                if host_group_id == storage_host_group_id:
+                                    hosts_list.append(host_id)
+                                    storage_host_group_relation = {
+                                        'storage_id': storage_id,
+                                        'native_storage_host_group_id':
+                                            storage_host_group_id,
+                                        'native_storage_host_id': host_id
+                                    }
+                                    storage_host_grp_relation_list.\
+                                        append(storage_host_group_relation)
+                    host_group_map = {
+                        "name": host_group.get('name'),
+                        "description": host_group.get('durable-id'),
+                        "storage_id": storage_id,
+                        "native_storage_host_group_id": storage_host_group_id,
+                        "storage_hosts": ','.join(hosts_list)
+                    }
+                    host_group_list.append(host_group_map)
+                storage_host_groups_result = {
+                    'storage_host_groups': host_group_list,
+                    'storage_host_grp_host_rels':
+                        storage_host_grp_relation_list
+                }
+            return storage_host_groups_result
+        except Exception as e:
+            LOG.error("Failed to get host_groups from msa")
```

Comment:
Log info should contain storage info, such as `storage_id` , so we can know which storage is going wrong.

Reply:
Error prompt statement changed
![image](https://user-images.githubusercontent.com/88076879/155677153-2a4e87f0-ae0e-407e-b94f-2a518e978a32.png)

Reply:
Error prompt statement changed
![image](https://user-images.githubusercontent.com/88076879/155677240-cc2f5339-a65b-4bbd-912c-36fde510d55f.png)